doc/development/plugins.rst

```diff
@@ -97,169 +92,165 @@ as well as potential further capabilities, by providing the following class attr
-
-  This is used to decide whether an observable is supported by your device in the
-  default implementation of the public method :meth:`.Device.supports_observable`.
+  See :doc:`/introduction/operations` for a full list of operations
+  supported by PennyLane.
 
 * :attr:`.Device._capabilities`: a dictionary containing information about the capabilities of
   the device. Keys currently supported include:
 
   * ``'model'`` (*str*): either ``'qubit'`` or ``'CV'``.
 
-  * ``'tensor_observables'`` (*bool*): ``True`` if the device supports measuring tensor products
-    of observables, ``False`` otherwise.
-
-For a better idea of how to best implement :attr:`.Device.operations` and
-:attr:`.Device.observables`, refer to the two reference plugins.
+  * ``"inverse_operations"`` (*bool*): ``True`` if the device supports
+    applying the inverse of operations. Operations which should be inverted
+    have ``operation.inverse == True``.
 
 .. note::
 
-    When writing a plugin device for PennyLane, make sure that your plugin
-    supports as many of the PennyLane built-in operations as possible.
+    If your device does not natively support an operation that has the
+    :meth:`~.Operation.decomposition` static method defined, PennyLane will
+    attempt to decompose the operation before calling the device. For example,
+    the :class:`~.Rot` `decomposition method <../_modules/pennylane/ops/qubit.html#Rot.decomposition>`_ will
+    decompose the single-qubit rotation gate to :class:`~.RZ` and :class:`~.RY` gates.
+
+
+.. note::
 
     If the convention differs between the built-in PennyLane operation
     and the corresponding operation in the targeted framework, ensure that the
     conversion between the two conventions takes place automatically
     by the plugin device.
 
-Device execution
-----------------
+Adding arguments to your device
+--------------------------------
 
-Once all the class attributes are defined, it is necessary to define some required class
-methods, to allow PennyLane to apply operations and measure observables on your device.
+Defining the ``__init__.py`` method of a custom device is not necessary; by default,
+the :class:`~.QubitDevice` initialization will be called, where the user can pass the
+following arguments:
 
-Applying operations
-^^^^^^^^^^^^^^^^^^^
+* ``wires`` (*int*): the number of wires on the device.
 
-To execute operations on the device, the following method must be defined:
+* ``shots=1000`` (*int*): number of circuit evaluations/random samples used to estimate
+  expectation values of observables in non-analytic mode.
 
-* :meth:`.Device.apply`: This accepts an operation name (as a string), the wires (subsystems)
-  to apply the operation to, and the parameters for the operation, and should apply the
-  resulting operation to given wires of the device.
+* ``analytic=True`` (*bool*): If ``True``, the device calculates probability, expectation
```

Comment:
Do we need to say that hardware is always using `analytic=False`?

Reply:
done!